L23-node-protobufjs-library.md

```diff
@@ -0,0 +1,72 @@
+
+The Node gRPC library currently depends on Protobuf.js version 5. Since Protobuf.js 6 was introduced, Protobuf.js 5 has not been maintained. In the past we have tried to upgrade that dependency in place, but that caused breaking changes that we did not accept. Our solution to this problem is to publish a new library that is solely responsible for loading `.proto` files using Protobuf.js and transforming them into a generic format that gRPC can use. The introduction of this new library provides an opportunity to improve the API for loading files based on feedback on the existing library.
+
+## Proposal
+
+For the purpose of definiing services in general, the existing gRPC library has defined the types [`MethodDefinition`](https://grpc.io/grpc/node/grpc.html#~MethodDefinition__anchor) and [`ServiceDefinition`](https://grpc.io/grpc/node/grpc.html#~ServiceDefinition__anchor). A `MethodDefinition` is an object with the following properties:
+
+ - `path`: The URL path for accessing this method
+ - `requestStream`: A boolean indicating whether there is a stream of request messages instead of a single message.
+ - `responseStreams`: A boolean indicating whether there is a stream of response messages instead of a single message.
+ - `requestSerialize`: A function for serializing a request object to a `Buffer`
+ - `responseSerialize`: A function for serializing a response object to a `Buffer`
+ - `requestDeserialize`: A function for deserializing a request object from a `Buffer`
+ - `responseDeserialize`: A function for deserializing a response object from a `Buffer`
+
+This information is sufficient to implement a client or service handler for any kind of method. Then a `ServiceDefintion` is just an object that maps method names to `MethodDefinition` objects. When loading a `.proto` file, more than one service may be loaded, so we additionally define a wrapping type `PackageDefinition`, which maps fully qualified service names to `ServiceDefinition` objects. For example, if a `.proto` file defines the package `a.b` with the services `Service1` and `Service2`, the resulting `PackageDefinition` would have the keys `"a.b.Service1"` and `"a.b.Service2"`.
+
+The library exposes the following functions:
+
+```js
+/**
+ * Asynchronously load a .proto file and all of its transitive imports
+ * @param {string} filename File path to the .proto file to load.
+ * @param {Object} options Options for loading the file and setting up the deserializers
+ * @return {Promise<PackageDefinition>} A promise for an object containing definitions for all of the loaded services
+ */
+load(filename, options)
+
+/**
+ * Synchronously load a .proto file and all of its transitive imports
+ * @param {string} filename File path to the .proto file to load.
+ * @param {Object} options Options for loading the file and setting up the deserializers
+ * @return {PackageDefinition} An object containing definitions for all of the loaded services
+ */
+loadSync(filename, options)
```

Comment:
Is this necessary to have? Can't we remove this util there is a real use case to add this (which won't be a breaking change).

Reply:
You're right. I will remove it.